### PR TITLE
fix(platform-server): correct dataloader return type

### LINKED ISSUE #207: correct dataloader return type
The normal returns of loader should be `T | null` or `T | Error`, but in the very beginning, for convenience, we made it only `T` which introduced some bugs we've met

### PR DESCRIPTION
**Pull request by lizimeow**
close #207

### CODE REVIEW

File: packages/platform-server/src/utils/data-loader.ts [hunk 1]
```diff
@@ -45,6 +45,6 @@ function normalizeBatchResults<K, V>(keys: readonly K[], cacheKey: keyof V | ((v
       indexedResults.set(key, res)
     })
 
-    return keys.map((key) => indexedResults.get(key)!)
+    return keys.map((key) => indexedResults.get(key))
```
Comment (forehalo) [inline packages/platform-server/src/utils/data-loader.ts:48]:
```suggestion
    return keys.map((key) => indexedResults.get(key) || new Error(`No entity with key ${key} found`))
```

File: packages/platform-server/src/modules/artifact/resolver.ts [hunk 2]
```diff
@@ -179,6 +179,6 @@ export class ArtifactEntrypointResolver {
   @ResolveField(() => Int)
   async artifactId(@Parent() entrypoint: ArtifactEntrypoint) {
     const artifact = await this.artifact.loader.load(entrypoint.artifactId)
-    return artifact.iid
+    return artifact?.iid
```
Comment (forehalo) [inline packages/platform-server/src/modules/artifact/resolver.ts:182]: it's not safe, the graphql return type is non-nullable

Comment (lizimeow) [reply packages/platform-server/src/modules/artifact/resolver.ts:182]: so, should i change it to nullable or throw error.

Comment (forehalo) [reply packages/platform-server/src/modules/artifact/resolver.ts:182]: nullable

**Review by forehalo — APPROVED** [review]

**Pull request merged**